be/src/vec/aggregate_functions/aggregate_function_collect.cpp

```diff
@@ -27,21 +27,25 @@
 
 namespace doris::vectorized {
 
-template <typename T, typename HasLimit>
+template <typename T, typename HasLimit, typename ShowNull>
 AggregateFunctionPtr do_create_agg_function_collect(bool distinct, const DataTypes& argument_types,
                                                     const bool result_is_nullable) {
-    if (distinct) {
-        return creator_without_type::create<
-                AggregateFunctionCollect<AggregateFunctionCollectSetData<T, HasLimit>, HasLimit>>(
+    if (ShowNull::value) {
```

Comment:
You might consider that `array_agg` is exactly the same as `collect_list` when the argument is non-nullable, and nothing needs to be changed.

Reply:
as below:
//ShowNull is just used to support array_agg because array_agg needs to display NULL
//todo: Supports order by sorting for array_agg